worldengine/draw.py

```diff
@@ -228,25 +230,26 @@ def average_colors(c1, c2):
 
     return (r, g, b)
 
+
 def get_normalized_elevation_array(world):
     ''' Convert raw elevation into normalized values between 0 and 255,
         and return a numpy array of these values '''
 
-    e = world.elevation['data']
+    e = world.layers['elevation'].data
 
-    mask = numpy.ma.array(e, mask=world.ocean)  # only land
+    mask = numpy.ma.array(e, mask=world.layers['ocean'].data)  # only land
     min_elev_land = mask.min()
     max_elev_land = mask.max()
     elev_delta_land = max_elev_land - min_elev_land
 
-    mask = numpy.ma.array(e, mask=numpy.logical_not(world.ocean))  # only ocean
+    mask = numpy.ma.array(e, mask=numpy.logical_not(world.layers['ocean'].data))  # only ocean
     min_elev_sea = mask.min()
     max_elev_sea = mask.max()
     elev_delta_sea = max_elev_sea - min_elev_sea
 
     c = numpy.empty(e.shape, dtype=numpy.float)
-    c[numpy.invert(world.ocean)] = (e[numpy.invert(world.ocean)] - min_elev_land) * 127 / elev_delta_land + 128
-    c[world.ocean] = (e[world.ocean] - min_elev_sea) * 127 / elev_delta_sea
+    c[numpy.invert(world.layers['ocean'].data)] = (e[numpy.invert(world.layers['ocean'].data)] - min_elev_land) * 127 / elev_delta_land + 128
+    c[world.layers['ocean'].data] = (e[world.layers['ocean'].data] - min_elev_sea) * 127 / elev_delta_sea
```

Comment:
At this point I would suggest introducing a reference to the ocean at the very top of the function; it will make thinks more readable.